File: src/main/java/org/featx/spec/entity/AbstractRecord.java
package org.featx.spec.entity;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;
import org.featx.spec.model.Record;

import java.time.LocalDateTime;

/**
 * @author Excepts
 * @since 2019/10/27 17:22
 */

@Data
@EqualsAndHashCode(callSuper = true)
@ToString(callSuper = true)
public abstract class AbstractRecord<I> extends AbstractIdentifiable<I> implements Record {

    private static final long serialVersionUID = -63131834985445498L;

    private LocalDateTime createdAt;

}

File: src/main/kotlin/org/featx/jvmon/entity/AbstractRecord.kt
package org.featx.jvmon.entity

import lombok.Data
import lombok.EqualsAndHashCode

import java.time.LocalDateTime

@Data
@EqualsAndHashCode(callSuper = true)
abstract class AbstractRecord<I> : AbstractIdentifiable<I>(), Record<I> {
    override var createdAt: LocalDateTime? = null
        set(createdAt) {
            field = this.createdAt
        }
}
